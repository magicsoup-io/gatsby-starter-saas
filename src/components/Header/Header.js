import { Link } from 'gatsby'
import PropTypes from 'prop-types'
import React, { Fragment } from 'react'

import { Heading, Container, Box, Flex, Link as MagicLink } from '@magicsoup.io/stock'
import { Menu, P } from '../../styled'
import { isLoggedIn, logout } from '../../services/auth'
import Status from '../Status';

const Header = ({ siteTitle, props }) => (
  <Menu>
    <Container width='100%'>
      <Flex 
        justifyContent='space-between'
        flexWrap='wrap'
        alignItems='center'>
        <Heading as='h1' variant='h1' style={{fontSize: 25}}>
          <Link
            to='/'
            style={{
              color: `white`,
              textDecoration: `none`,
            }}
          >
            {siteTitle}
          </Link>
        </Heading>
        <P color='white' mt={0}>
          <Link to='/' style={{color: '#d3d2d1', paddingRight: '20px' }}>Home</Link>
          <Link to='/app/plans' style={{color: '#d3d2d1', paddingRight: '20px' }}>Pricing</Link>
          <Link to='/app/profile' style={{color: '#d3d2d1', paddingRight: '20px' }}>Profile</Link>
        </P>
        <Box mb={[2,0]}><Status color='white' linkColor='#d3d2d1' /></Box>
      </Flex>
    </Container>
  </Menu>
)

Header.propTypes = {
  siteTitle: PropTypes.string,
}

Header.defaultProps = {
  siteTitle: ``,
}

export default Header
