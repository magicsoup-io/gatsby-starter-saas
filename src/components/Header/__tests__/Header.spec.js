import React from "react"
import renderer from "react-test-renderer"
import { StaticQuery } from 'gatsby'
import Header from "../Header"

describe("Header", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Header siteTitle="Seitentitel" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})