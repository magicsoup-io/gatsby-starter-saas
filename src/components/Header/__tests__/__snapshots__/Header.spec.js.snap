// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<nav
  className="sc-bdVaJa sc-gZMcBi gCoWvL"
>
  <div
    className="sc-bdVaJa sc-bxivhb djuapn"
    width="100%"
  >
    <div
      className="sc-bdVaJa sc-ifAKCX WvoHu"
    >
      <h1
        className="sc-bdVaJa sc-EHOje sc-bZQynM kKcxZB"
        style={
          Object {
            "fontSize": 25,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Seitentitel
        </a>
      </h1>
      <p
        className="sc-bdVaJa sc-EHOje sc-gqjmRU hmgTND"
        color="white"
        fontSize={2}
      >
        <a
          href="/"
          style={
            Object {
              "color": "#d3d2d1",
              "paddingRight": "20px",
            }
          }
        >
          Home
        </a>
        <a
          href="/app/plans"
          style={
            Object {
              "color": "#d3d2d1",
              "paddingRight": "20px",
            }
          }
        >
          Pricing
        </a>
        <a
          href="/app/profile"
          style={
            Object {
              "color": "#d3d2d1",
              "paddingRight": "20px",
            }
          }
        >
          Profile
        </a>
      </p>
      <div>
        <p
          className="sc-bdVaJa sc-EHOje sc-gqjmRU hmgTND"
          color="white"
          fontSize={2}
        >
          To get the full app experience, youâ€™ll need to
           
          <a
            href="/app/login"
            onClick={[Function]}
            style={
              Object {
                "color": "#d3d2d1",
              }
            }
          >
            log in
          </a>
          .
        </p>
      </div>
    </div>
  </div>
</nav>
`;
